src/core/structure/sheetHolder.ts

```diff
@@ -0,0 +1,49 @@
+import LightSheet from "../../main.ts";
+import { CellKey, SheetKey } from "./key/keyTypes.ts";
+import Cell from "./cell/cell.ts";
+
+export default class SheetHolder {
+  sheets: Map<SheetKey, LightSheet>;
+  sheetNames: Map<string, SheetKey>;
+  cellData: Map<CellKey, Cell>;
+
+  constructor() {
+    this.sheets = new Map();
+    this.sheetNames = new Map();
+    this.cellData = new Map();
+
+    if (window.sheetHolder) {
+      return window.sheetHolder;
+    }
+    window.sheetHolder = this;
+  }
```

Comment:
I would private the contractor and create a static function called something like `GetOrCreateSheetHolder`, which, as you might have guessed, checks if the Holder exist, if not, call the private constructor which creates `sheetHolder` normally. 

In the current state, it is extremely unexpected that the constructor doesn't actually create a new instance.